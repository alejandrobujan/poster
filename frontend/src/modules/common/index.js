export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";