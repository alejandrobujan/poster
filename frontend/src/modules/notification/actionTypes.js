export const FIND_NOTIFICATIONS_COMPLETED = "poster/notifications/findNotificationsCompleted";
export const SET_VIEWED = "poster/notifications/setViewed";