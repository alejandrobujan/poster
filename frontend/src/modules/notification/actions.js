import * as actionTypes from './actionTypes';
import backend from '../../backend';

export const findNotifications = () => dispatch => {

    backend.notificationService.findNotifications(
        notifications => dispatch(findNotificationsCompleted(notifications)));

}
const findNotificationsCompleted = notifications => ({
    type: actionTypes.FIND_NOTIFICATIONS_COMPLETED,
    notifications
});