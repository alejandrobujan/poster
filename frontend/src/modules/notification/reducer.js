import { combineReducers } from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
	notifications: [],
};

const notifications = (state = initialState.notifications, action) => {

	switch (action.type) {

		case actionTypes.FIND_NOTIFICATIONS_COMPLETED:
			return action.notifications;

		case actionTypes.SET_VIEWED:
			return [...(state).map(n => (n.notificationId === action.notificationId ? { ...n, notificationViewed: true } : n))];

		default:
			return state;

	}

}


const reducer = combineReducers({
	notifications
});

export default reducer;