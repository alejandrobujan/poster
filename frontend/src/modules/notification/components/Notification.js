import { getDate } from '../../../backend/utils';
import { useDispatch } from 'react-redux';
import * as actions from '../actions';
import { useNavigate } from 'react-router-dom';

const Notification = ({ id, text, viewed, creationDate, postId, notifierUserDto }) => {
	
	const dispatch = useDispatch();
	const navigate = useNavigate();
	
	const handleNotificationClick = () => {
		dispatch(actions.markAsViewed(id));
		navigate(`/post/post-details/${postId}`);
	}

	return (
		<button data-testid="NotificationButton" className="dropdown-item border-bottom" style={{ overflowX: 'hidden' }} onClick={handleNotificationClick}>
			<div className='' style={{minWidth: '300px'}}>
				<div className="row text-center">
					<div className="col-2">
						<img data-testid="Avatar" src={notifierUserDto.avatar ? `data:image/*;base64,${notifierUserDto.avatar}` : "/poster/assets/profile.png"} alt="Avatar" height="30px" width="30px" />
					</div>
					<div className="col-8 text-center">
						<small style={{ maxWidth: '100%', display: 'block', whiteSpace: 'normal', fontSize: '12px' }}>
							<span data-testid="Date" style={{ color: 'grey', display: 'block', whiteSpace: 'nowrap' }}>{getDate(creationDate)}</span><br />
						</small>
					</div>
					<div className="col-2">
						{!viewed && <span data-testid="WarningNotification" className="badge bg-danger text-white rounded-pill"><small style={{color:'#dc3545'}}>&nbsp;â€¢</small></span>}
					</div>
				</div>
				<span data-testid="NotificationText" style={{ maxWidth: '100%', display: 'block', whiteSpace: 'normal', fontSize: '12px', textAlign: 'justify' }}>{text}</span>
			</div>
		</button>
	)
};

export default Notification;