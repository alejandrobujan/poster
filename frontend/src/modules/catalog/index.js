import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


