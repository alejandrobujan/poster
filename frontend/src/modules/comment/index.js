import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};