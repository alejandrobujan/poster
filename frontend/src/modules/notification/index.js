import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Notifications} from "./components/Notifications";
export {default as Notification} from "./components/Notification";

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


