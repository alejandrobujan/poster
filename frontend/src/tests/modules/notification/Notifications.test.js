import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';

import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import renderer from "react-test-renderer";
import { notificationMock } from "../../state/Notification.mock";

import { Notifications } from "../../../modules/notification";

describe("Notifications", () => {

	const middlewares = [thunk];
	const mockStore = configureMockStore(middlewares);
	let store;
	store = mockStore(notificationMock);

	const emptyNotifications =[];

	const notifications =
		[
			{
				notificationId: 1,
				notificationText: "Hola",
				notificationViewed: false,
				notificationCreationDate: null,
				postId: 1,
				notifierUserDto: {
					avatar: null
				}
			}
		];


	it("renders correctly", () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<MemoryRouter>
						<Notifications notifications={notifications} />
					</MemoryRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("basic structure no notifications", async () => {
		render(
			<Provider store={store}>
				<MemoryRouter>
					<Notifications notifications={emptyNotifications} />
				</MemoryRouter>
			</Provider>
		);

		const noNotifications = screen.getByTestId('NoNotifications');

	});

	it("basic structure notifications", async () => {
		render(
			<Provider store={store}>
				<MemoryRouter>
					<Notifications notifications={notifications} />
				</MemoryRouter>
			</Provider>
		);

		const notificationsScreen = screen.getByTestId('Notifications');

	});

});