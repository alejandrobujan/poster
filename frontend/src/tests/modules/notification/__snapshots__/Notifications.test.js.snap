// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications renders correctly 1`] = `
<li
  className="nav-item dropdown"
  data-testid="Notifications"
>
  <a
    className="dropdown-toggle nav-link"
    data-toggle="dropdown"
    href="#!"
  >
    <svg
      className="bi bi-bell"
      fill="currentColor"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 16a2 2 0 0 0 2-2H6a2 2 0 0 0 2 2zM8 1.918l-.797.161A4.002 4.002 0 0 0 4 6c0 .628-.134 2.197-.459 3.742-.16.767-.376 1.566-.663 2.258h10.244c-.287-.692-.502-1.49-.663-2.258C12.134 8.197 12 6.628 12 6a4.002 4.002 0 0 0-3.203-3.92L8 1.917zM14.22 12c.223.447.481.801.78 1H1c.299-.199.557-.553.78-1C2.68 10.2 3 6.88 3 6c0-2.42 1.72-4.44 4.005-4.901a1 1 0 1 1 1.99 0A5.002 5.002 0 0 1 13 6c0 .88.32 4.2 1.22 6z"
      />
    </svg>
    <span
      className="badge bg-danger text-white ms-1 rounded-pill"
    >
      1
    </span>
  </a>
  <div
    className="dropdown-menu dropdown-menu-right notification-box"
  >
    <button
      className="dropdown-item border-bottom"
      data-testid="NotificationButton"
      onClick={[Function]}
      style={
        Object {
          "overflowX": "hidden",
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "minWidth": "300px",
          }
        }
      >
        <div
          className="row text-center"
        >
          <div
            className="col-2"
          >
            <img
              alt="Avatar"
              data-testid="Avatar"
              height="30px"
              src="/poster/assets/profile.png"
              width="30px"
            />
          </div>
          <div
            className="col-8 text-center"
          >
            <small
              style={
                Object {
                  "display": "block",
                  "fontSize": "12px",
                  "maxWidth": "100%",
                  "whiteSpace": "normal",
                }
              }
            >
              <span
                data-testid="Date"
                style={
                  Object {
                    "color": "grey",
                    "display": "block",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                1/1/1970 12:00:00 AM
              </span>
              <br />
            </small>
          </div>
          <div
            className="col-2"
          >
            <span
              className="badge bg-danger text-white rounded-pill"
              data-testid="WarningNotification"
            >
              <small
                style={
                  Object {
                    "color": "#dc3545",
                  }
                }
              >
                 •
              </small>
            </span>
          </div>
        </div>
        <span
          data-testid="NotificationText"
          style={
            Object {
              "display": "block",
              "fontSize": "12px",
              "maxWidth": "100%",
              "textAlign": "justify",
              "whiteSpace": "normal",
            }
          }
        >
          Hola
        </span>
      </div>
    </button>
  </div>
</li>
`;
